File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = s.size();
        int m = t.size();
        
        if(s == t) return true;
        if(n != m) return false;
        
        vector<int> freq1(26,0);
        vector<int> freq2(26,0);
        
        for(int i=0; i<n; i++){
            freq1[s[i]-'a']++;
            freq2[t[i] - 'a']++;
        }
        
        return (freq1 == freq2);
    }
};

File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m = len(s)
        n = len(t)
        
        if m != n: return False
        
        freq = [0]*26
        
        for c in s:
            freq[ord(c) -ord('a')] += 1
        
        for c in t:
            freq[ord(c)-ord('a')] -= 1
            
        return True if max(freq) == 0 else False
            
        